lib/features/artifacts.js

```diff
@@ -128,7 +128,22 @@ export default class Artifacts {
       };
 
       try {
-        await uploadArtifact(taskHandler, stream, entryName, expiry, headers);
+        await new Promise((accept, reject) => {
+          let numRetries = taskHandler.runtime.retry.numRetries;
+          let retry = () => {
+            uploadArtifact(taskHandler, stream, entryName, expiry, headers)
```

Comment:
So I'm not 100% why this is working.  My understanding is that once it reads from the stream once, that data is drained from the stream so subsequent calls to upload the artifact and read from the same stream *should* fail because the data isn't there.  It might be because this file is below the watermark threshold for the stream (I think 16k).  Perhaps we could have a test that's uploading a directory containing multiple files of multiple sizes and verifying the contents of those files after upload.